signer/src/stacks/contracts.rs

```diff
@@ -289,7 +289,7 @@ pub struct CompleteDepositV1 {
     /// above UTXO because of bitcoin mining fees.
     pub amount: u64,
     /// The address where the newly minted sBTC will be deposited.
-    pub recipient: PrincipalData,
+    pub recipient: Box<PrincipalData>,
```

Comment:
Rather than boxing the values here, why not box the `ContractCall` type in the `StacksTx::ContractCall` variant? In general, it seems preferable to doing that than boxing individual struct fields.

Reply:
Changed it in [013b328](https://github.com/stacks-sbtc/sbtc/pull/1679/commits/013b328318fdbf7aa588f2d13c67f43c4e358195). Yeah, it's probably better to box the whole thing so we're not always allocating the largest size for every instance.

Reply:
Oh, I actually meant do:
```rust
#[derive(Clone, Debug, Hash, PartialEq)]
pub enum StacksTx {
    /// A contract call transaction.
    ContractCall(Box<ContractCall>), // <-- this guy
    /// A smart contract transaction used for deploying smart contract.
    SmartContract(SmartContract),
}
```
But your change also works, and leads to a similar diff. 👍🏽

Reply:
Ah, that wouldn't solve the issue though because you'd still have an enum (`ContractCall`) with large/unbalanced variants that can be independently instantiated, which I assume is why clippy explicitly warns on `ContractCall` and not its "users".